tests/test_main.py

```diff
@@ -1,29 +1,35 @@
 import os
+import typing
 
 from click.testing import CliRunner
 
 import httpx
+from httpx import main
 
+if typing.TYPE_CHECKING:  # pragma: no cover
+    # don't let mypy be misled by the fallback defined in httpx/__init__.py
+    from httpx._main import main  # noqa: F811
```

Comment:
I like what you've done here - we've got other cases where the `TYPE_CHECKING` imports from private namespaces, so I think this is okay.